kernel/src/kernel.rs

```diff
@@ -92,27 +82,18 @@ fn try_allocate_grant(driver: &dyn SyscallDriver, process: &dyn process::Process
     }
 }
 
-struct KernelProcessInitCapability {}
-unsafe impl capabilities::ProcessInitCapability for KernelProcessInitCapability {}
-
-struct KernelProcessApprovalCapability {}
-unsafe impl capabilities::ProcessApprovalCapability for KernelProcessApprovalCapability {}
-
 impl Kernel {
+    /// Create the kernel object that knows about the list of processes.
+    ///
+    /// Crucially, the processes included in the `processes` array MUST be valid
```

Comment:
I think this is violated everywhere. It looks to me like the kernel is created with a typically empty `PROCESSES` list and is filled by the loader/checker afterwards. Perhaps this means something different that how I'm interpreting the comment?

Reply:
`PROCESSES` is `[Option<&'static Process>]`. So a `[None, None, None, ...]` has no processes. A "process included in the array" is a `Some(&ProcessStandard)`.

Reply:
OK, I misunderstood the comment. This is just saying that processes are _assumed_ to be valid to run as soon as they appear, and that this component (the `Kernel`) does no validation of any kind.